shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1305,6 +1316,16 @@ - (void)keyboardWillChangeFrame:(NSNotification*)notification {
   } else {
     self.targetViewInsetBottom = 0;
   }
+
+  bool keyboardIsShowing = beginKeyboardFrame.origin.y > keyboardFrame.origin.y;
+
+  // Flag for simultaneous compounding animation calls.
+  self.keyboardAnimationIsCompounding = (self.keyboardAnimationIsShowing && keyboardIsShowing) ||
+                                        (!self.keyboardAnimationIsShowing && !keyboardIsShowing);
+
+  // Mark keyboard as showing or hiding
+  self.keyboardAnimationIsShowing = keyboardIsShowing;
+
```

Comment:
1: Using `BOOL` instead of `bool`

2:
maybe `keyboardIsShowing` can renamed with `keyboardWillShow`

3:
```obj-c
self.keyboardAnimationIsCompounding = (self.keyboardAnimationIsShowing && keyboardIsShowing) ||
                                        (!self.keyboardAnimationIsShowing && !keyboardIsShowing);
```
Can be replaced with :
```objc
self.keyboardAnimationIsCompounding = self.keyboardAnimationIsShowing == keyboardIsShowing
```

4: Add "." for comment

(*^▽^*)